### DISCUSSION TITLE
Expected behaviour? Fixed model version in OSF links?

### DISCUSSION THREAD

**Discussion by mathieuboudreau** [neuropoly/axondeepseg#372] [Bug reports, suggestions for code refactoring]
https://github.com/neuropoly/axondeepseg/blob/30db780dd3479abe611bc69258ec180b14e2486b/AxonDeepSeg/models/download_model.py#L15-L16

@mariehbourget can you confirm that we do in fact want to have the fixed versions in the links above? i.e. even though we update the models on OSF, we need to manually update this in the code was well. If so, is this documented anywhere on OSF and/or GitHub?

**Comment by mariehbourget** [comment]
That is a good question. It is in fact documented on wiki [here](https://github.com/neuropoly/axondeepseg/wiki/Instructions-on-How-to-Update-Models-in-Repo-and-OSF-Storage).

However, the version does not appear in the Readme file, as it is suggested in the wiki. It only appears in the last line of the template for the Readme [here](https://github.com/neuropoly/axondeepseg/wiki/Template-for-Readme.txt) which uses `_vX` for versioning at the end of the name of the model.

Since we now use the versioning of OSF and the same name (without `_vX`) for any version of the model, I don't see why we would have to have fixed versions in the links above. When installing ADS, we want users to download the latest version of the models and avoid errors with manual updates of the code.

I would suggest to remove the fixed versions from `download_model.py` and to update the wiki accordingly.
@mathieuboudreau, do you see other reasons for keeping the fixed versions? If not, do you want me to open a new PR for this correction or to make the correction in the PR #373 ?

**Comment by mathieuboudreau** [comment]
Thanks @mariehbourget ! I think the reason why maybe we've kept the fixed versions, is that it makes it easier to test a new model in a branch (upload it to OSF and change the version number in your branch, this way master doesn't get affected). So if we don't have the fixed versions, it's likely that we could break the software by simply uploading a bad model. But on the other hand, we might forget to update the version number after uploading a new model too, so...

Really not sure what the best solution might be here, both have benefits and faults. I think clear documentation and workflows are needed for either one.

**Comment by mariehbourget** [comment]
Good point. The question is: do we plan on uploading models on OSF for testing purposes only?
I kind of thought that we would test locally and only replace the default models folder on OSF with a new, better, tested and approved version.

Like you, I'm not sure what the best solution is.

However in both cases, changing the version on OSF can be tricky for users if they started an experiment with one version, update ADS for whatever reasons, and then end up with a new version for the remaining of their experiment.

**Comment by jcohenadad** [comment]
> Since we now use the versioning of OSF and the same name (without _vX) for any version of the model, I don't see why we would have to have fixed versions in the links above. When installing ADS, we want users to download the latest version of the models and avoid errors with manual updates of the code.

in addition to Mathieu's argument https://github.com/neuropoly/axondeepseg/issues/372#issuecomment-717373803, wasn't the purpose to also ensure retro-compatibility? E.g.: If a user wants to re-install an older version of ADS(*), if we don't specify the version of the model, the latest will be downloaded, which will produce different results than results that used to be produced with that old version of ADS.

(*): e.g. because they did processing 6 months ago, and now the reviewer of the submitted article is asking for an additional processing-- or simply to ensure reproducibility of published results

Also, let's keep in mind that ultimately, once ADS will be calling ivadomed, models will be hosted on GH and will follow versioning there. Our current approach (in SCT, ivadomed, shimming-toolbox) is to specify the link to the release artifact on the data manager (here: download_model), so yes: it requires diligence and proper documentation. But so far, things are going smoothly with that approach, and I do see more pros than cons.

**Comment by mariehbourget** [comment]
Thanks @jcohenadad, those are strong pros. We should then keep the version in download_models which is already documented in wiki. 
I will still make corrections to the [wiki](https://github.com/neuropoly/axondeepseg/wiki/Instructions-on-How-to-Update-Models-in-Repo-and-OSF-Storage) to remove point 2, and to the [template](https://github.com/neuropoly/axondeepseg/wiki/Template-for-Readme.txt) to remove the `_vX` part in the filename, those instructions are not up-to-date since the version is no longer indicated in the filename.

**Comment by mariehbourget** [comment]
We decided to keep the fixed versions in `download_models.py`
I made the corrections mentioned in the comment above in the wiki, so I'm closing this issue.
